CoreFoundation/Base.subproj/CFPlatform.c

```diff
@@ -1416,8 +1434,15 @@ void _CFThreadSpecificSet(_CFThreadSpecificKey key, CFTypeRef _Nullable value) {
 #if TARGET_OS_WIN32
     if (value != NULL) {
         swift_retain((void *)value);
+
+        _CFThreadSpecificData *data = malloc(sizeof(_CFThreadSpecificData));
+        data->value = value;
+        data->key = key;
+
+        FlsSetValue(key, data);
+    } else {
+        FlsSetValue(key, NULL);
```

Comment:
This may cause a leak, and worse a crash - you may have had an entry and you lost it.

Reply:
Oh, this is definitely a leak. Thanks!

I wonder how does it work for Swift value we store here? We are not doing `swift_release` for previously stored object when overwriting it. But it not leaks, as far as I see.

Reply:
This isn't resolved.  The problem is that you are now injecting a value that you are de-referencing rather than previously which was passed to `free` or `swift_release` which accept and ignore `NULL`.

Reply:
You're right, by bad :(

Reply:
Done